### PR TITLE
adapt to OOT MCM

### PR DESCRIPTION
**Pull request by kon-angelo**
**How to categorize this PR?**

/area control-plane
/kind api-change
/priority normal
/platform openstack

**What this PR does / why we need it**:
Changes the MCM deployment to the OOT repo one. Also adapts the `MachineClass` templates for the new CR.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:
Drafted until first release of the OOT repo. 

**Release note**:

```other operator
Openstack extension now uses the new Out-Of-Tree MCM implementation.
```

### CODE REVIEW

**Comment by prashanth26** [comment]
Hi @kon-angelo ,

Thanks for opening this up. I will review this by Thursday. Apologies for the delay.

/assign

File: charts/images.yaml
Comment (prashanth26) [inline charts/images.yaml:38]: I thought this release version should have been, however, I see your release version matching `v0.0.1`.  You can take a final call on this. 
```suggestion
  tag: "v0.1.0"
```

File: charts/internal/machine-controller-manager/seed/templates/deployment.yaml
Comment (prashanth26) [inline charts/internal/machine-controller-manager/seed/templates/deployment.yaml:42]: I am not a logging/metrics expert, but changing the container names hopefully doesn't affect metric/log collection. If you aren't sure, i would suggest to revert it to `openstack-machine-controller-manager` and we can probably relook at metric improvements in a second round.

File: charts/internal/machine-controller-manager/seed/templates/deployment.yaml
Comment (prashanth26) [inline charts/internal/machine-controller-manager/seed/templates/deployment.yaml:95]: Please add the following timeout flags explicitly [here](https://github.com/gardener/gardener-extension-provider-aws/pull/212/files#diff-e38adde4c814c39f0c3443b395a76dfc43639b003e8514c731b39e2b1683aea9R48-R50)

File: charts/internal/machine-controller-manager/seed/templates/deployment.yaml
Comment (prashanth26) [inline charts/internal/machine-controller-manager/seed/templates/deployment.yaml:86]: Please add this flag for explicit migration from [here](https://github.com/gardener/gardener-extension-provider-aws/pull/212/files#diff-e38adde4c814c39f0c3443b395a76dfc43639b003e8514c731b39e2b1683aea9R77), as this would take care of cleaning up of `OpenstackMachineClass` in case of migration triggered from MCM end.

File: pkg/controller/worker/machines.go
Comment (prashanth26) [inline pkg/controller/worker/machines.go:37]:
```suggestion
// MachineClassKind yields the name of the machine class kind used by Openstack provider.
```

File: pkg/controller/worker/machines.go
Comment (prashanth26) [inline pkg/controller/worker/machines.go:47]:
```suggestion
// MachineClassList yields a newly initialized MachineClassList object.
```

File: charts/images.yaml
Comment (kon-angelo) [reply charts/images.yaml:38]: The idea was to release `v0.0.1` as an alpha version to validate on dev and have the leeway to release `v0.1.0` if there is a need to do fixes after validation.

**Comment by kon-angelo** [comment]
@prashanth26 I updated the PR based on your comments. One extra change I had to make was add the `poddisruptionbudget` RBACs for mcm `v.037.0`

File: charts/images.yaml
Comment (prashanth26) [reply charts/images.yaml:38]: Okay sounds good. Thanks.

**Review by prashanth26 — DISMISSED** [review]
Thank you for the changes. I also hope you have tried to test most cases to make sure migration works as expected. A few simple cases having the following,

- [ ] Create a cluster without changes and make sure that the shoot goes through migration and then finally is deleted successfully. 
- [ ] Create and delete cluster with changes.
- [ ] Perform simple scale-up/down and upgrade of nodes using new changes. 

Thanks. 
/lgtm if these basic tests are done locally.

**Comment by kon-angelo** [comment]
Thanks for the review @prashanth26. 

I retested all the scenarios you suggested (create/delete/migrate/scaling) and things worked smoothly. I will go on and merge it now.

**Comment by prashanth26** [comment]
Thank you @kon-angelo . Yes sure.

File: charts/internal/machine-controller-manager/seed/templates/deployment.yaml
Comment (dkistner) [inline charts/internal/machine-controller-manager/seed/templates/deployment.yaml:86]: Why `Always` would be `IfNotPresent` also sufficient?

File: pkg/controller/worker/machines.go
Comment (dkistner) [inline pkg/controller/worker/machines.go:60]: I guess this code is just temporarily required? Can we add a TODO to remind us to remove it in a few releases?

File: charts/internal/machine-controller-manager/seed/templates/deployment.yaml
Comment (kon-angelo) [reply charts/internal/machine-controller-manager/seed/templates/deployment.yaml:86]: Nice catch. It was a development artifact.

**Review by dkistner — APPROVED** [review]
/lgtm

**Review by prashanth26 — APPROVED** [review]
/lgtm

**Pull request merged**
